Allow the next break or microbreak to be scheduled from the CLI

lkrms opened the pull request:
Issue: closes #1091

### Requirements

- [x]  issue was opened to discuss proposed changes before starting implementation. It is important do discuss changes before implementing them (Why should we add it? How should it work? How should it look? Where will it be? ...).
- [x]  during development, `node` version specified in `package.json` was used (ie using [nvm](https://github.com/creationix/nvm)).
- [x]  package versions and package-lock.json were not changed (`npm install --no-save`).
- [x]  app version number was not changed.
- [ ]  all new code has tests to ensure against regressions.
- [ ] `npm run lint` reports no offenses.
- [ ] `npm run test` is error-free.
- [ ]  README and CHANGELOG were updated accordingly.
- [ ]  after PR is approved, all commits in it are [squashed](https://gitbetter.substack.com/p/how-to-squash-git-commits)

### Description of the Change

In the `Command` class:
- Add new `--wait` option to commands `mini` and `long`.
- Add `waitToMs` method to utilise the existing interval parser (`parseDuration`) from `main.js` when processing CLI options.

`BreaksPlanner`:
- Rename `skipToMicrobreak` to `scheduleMicrobreak` and modify it slightly, adding a new `skipToMicrobreak` function that simply calls `scheduleMicrobreak(100)`
- Ditto for `skipToBreak` and `scheduleBreak`

- Fix a few minor typos and inconsistencies in logged output.

lkrms committed:
Allow the next break or microbreak to be scheduled from the CLI

`Command`:
- Add new `--wait` option to commands `mini` and `long`
- Add `waitToMs` method to utilise the existing interval parser,
  `parseDuration`

`BreaksPlanner`:
- Split `skipToMicrobreak` to add `scheduleMicrobreak`
- Same for `skipToBreak`, `scheduleBreak`

- Fix typos and inconsistencies in logged output

hovancik commented on app/breaksPlanner.js:
```diff
@@ -168,34 +168,42 @@ class BreaksPlanner extends EventEmitter {
     const shouldMicrobreak = this.settings.get('microbreak')
-    const breakInterval = this.settings.get('breakInterval') + 1
     if (shouldBreak && shouldMicrobreak) {
+      const breakInterval = this.settings.get('breakInterval') + 1
```
any reason why this line was moved?

lkrms replied:
1. For consistency with `skipToBreak` (I compared the functions to check for any variations in behaviour).
2. Because the call to `this.settings.get` is unnecessary unless `shouldBreak && shouldMicrobreak` is true.

hovancik commented on app/breaksPlanner.js:
```diff
@@ -168,34 +168,42 @@ class BreaksPlanner extends EventEmitter {
   }
 
-  skipToMicrobreak () {
+  scheduleMicrobreak (delay) {
```
I think we should keep this function as `skipToMicrobreak (delay = 100)` instead of having 2 functions `skipToMicrobreak` and `scheduleMicrobreak`.

lkrms replied:
No worries. That's how I wrote it initially, but then I thought you might prefer more explicit semantics. I'll revert to the original approach.

hovancik commented on app/main.js:
```diff
@@ -85,26 +85,39 @@ if (!gotTheLock) {
+      case 'mini': {
         log.info('Stretchly: skip to Mini Break (requested by second instance)')
+        const ms = cmd.waitToMs()
+        if (ms === -1) {
```
can we rename `ms` to `delay`? It would be better readable

lkrms replied:
No worries. I'll make the `pause` code clearer while I'm at it, given that's where I ~~stole~~ borrowed this from :)

hovancik commented on app/main.js:
```diff
@@ -942,27 +955,37 @@ function postponeBreak (shouldPlaySound = false) {
 }
 
-function skipToMicrobreak () {
+function skipToMicrobreak (delay = 0) {
```
No default value needed. (If we add it as `skipToMicrobreak (delay = 100)` in `BreaksPlanner`)

lkrms replied:
Indeed. Actually, this was superfluous even without that. Fixed.

hovancik commented on app/utils/commands.js:
```diff
@@ -20,6 +20,12 @@ const allOptions = {
     description: 'Do not skip directly to this break (Long or Mini)',
     withValue: false
   },
+  wait: {
```
Could you also please add an axample usage to `allExamples`?

hovancik reviewed (changes requested):
Thanks! I added some comments

lkrms committed:
Review break scheduling code

- breaksPlanner.js: remove separate `scheduleMicrobreak` and
  `scheduleBreak` functions
- main.js:
  - Give `ms` variables more descriptive names
  - Remove default `delay` from `skipToMicrobreak` and `skipToBreak`
- commands.js: add an example

lkrms commented:
I've made the requested changes in a new commit. To avoid losing context, I didn't rebase or squash, but am happy to if needed.

hovancik reviewed (changes requested):
Thanks for the changes! I've tried locally and all seems fine :) Could you please update the CHANGELOG and squash?

hovancik committed:
Update readme

hovancik committed:
Adds tests

lkrms commented:
> Thanks for the changes! I've tried locally and all seems fine :) Could you please update the CHANGELOG and squash? 

I'm sorry I didn't get to this sooner. FOSS contribution time has been difficult to find over the last couple of weeks, but I was happy to do it. Thank you for merging. 🙏🏻

hovancik commented:
No worries :) I know the feeling, on the same boat here
